src/collection-facets/more-facets-content.ts

```diff
@@ -0,0 +1,361 @@
+/* eslint-disable dot-notation */
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  nothing,
+  PropertyValues,
+} from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import {
+  Aggregation,
+  Bucket,
+  SearchServiceInterface,
+  SearchParams,
+} from '@internetarchive/search-service';
+import { CollectionNameCacheInterface } from '@internetarchive/collection-name-cache';
+import { SelectedFacets, defaultSelectedFacets } from '../models';
+import { LanguageCodeHandlerInterface } from '../language-code-handler/language-code-handler';
+import '@internetarchive/ia-activity-indicator/ia-activity-indicator';
+import './more-facets-pagination';
+
+@customElement('more-facets-content')
+export class FacetsMoreContent extends LitElement {
+  @property({ type: String }) facetKey?: string;
+
+  @property({ type: String }) facetAggregationKey?: string;
+
+  @property({ type: String }) fullQuery?: string;
+
+  @property({ type: Object }) modalManager?: any;
+
+  @property({ type: Object }) searchService?: SearchServiceInterface;
+
+  @property({ type: Object })
+  collectionNameCache?: CollectionNameCacheInterface;
+
+  @property({ type: Object })
+  languageCodeHandler?: LanguageCodeHandlerInterface;
+
+  @property({ type: Object }) selectedFacets?: SelectedFacets;
+
+  @state() aggregations?: Record<string, Aggregation>;
+
+  @state() castedBuckets?: Bucket[] = [];
+
+  @state() pageNumber = 1;
+
+  /**
+   * Facets are loading on popup
+   */
+  @state() facetsLoading = true;
+
+  @state() paginationSize = 0;
+
+  private facetsPerPage = 60; // Q. how many items we want to have on popup view
+
+  async updated(changed: PropertyValues) {
+    if (changed.has('facetKey')) {
+      this.facetsLoading = true;
+      this.pageNumber = 1;
+
+      await this.fetchSpecificFacets(
+        this.facetAggregationKey as unknown as string
+      );
+      this.facetsLoading = false;
+    }
+
+    if (changed.has('aggregations')) {
+      this.filterFacets();
+    }
+  }
+
+  async fetchSpecificFacets(facetAggregationKey: string) {
```

Comment:
suggestion: add a lil doc block above to note side effects. not only you are fetching but you are setting dynamic property `this.aggregations` below